include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp

```diff
@@ -60,20 +63,25 @@ class Earth : public Model
         enum class Type
         {
 
+            Spherical,          ///< The spherical gravity originating from a point source at the center of the Earth
             WGS84,              ///< The normal gravitational field for the reference ellipsoid. This includes the zonal coefficients up to order 20.
             EGM84,              ///< The Earth Gravity Model 1984, which includes terms up to degree 180.
             EGM96,              ///< The Earth Gravity Model 1996, which includes terms up to degree 360.
             EGM2008             ///< The Earth Gravity Model 2008, which includes terms up to degree 2190.
 
         } ;
 
-        /// @brief              Constructor
+        /// @brief              Constructor with max degree and order variables
         ///
         /// @param              [in] aType A gravitational model type
         /// @param              [in] (optional) aDataDirectory A gravitational model data directory
+        /// @param              [in] (optional) aGravityModelDegree A gravitational model degree
+        /// @param              [in] (optional) aGravityModelOrder A gravitational model order
 
                                 Earth                                       (   const   Earth::Type&                aType,
-                                                                                const   Directory&                  aDataDirectory                              =   Directory::Undefined() ) ;
+                                                                                const   Directory&                  aDataDirectory                              =   Directory::Undefined(),
+                                                                                const   Integer&                    aGravityModelDegree                         = -1,
+                                                                                const   Integer&                    aGravityModelOrder                          = -1    ) ;
```

Comment:
Instead of using the special value `-1` to convey that a value is undefined, use `Integer::Undefined()`.

Reply:
@lucas-bremond @Derollez There is a specific reason why I used -1 instead of an undefined integer, because Geographic lib interprets negative numbers a specific way and does not truncate the gravity model. If you inputted an undefined integer it would break how geographic lib interprets the numbers.

Reply:
https://github.com/geographiclib/geographiclib/blob/main/include/GeographicLib/GravityModel.hpp#L193